### PR TITLE
Added process module enumeration for OSx.

### PR DESCRIPTION
**Pull request by mikem8361**
Enable building dbi and dbgshim on OSx. Cache the process module list in the local process data because EnumProcessModules/GetModuleFileNameExW are called by dbgshim a lot and building the list every time really affected performance especially on OSx.

### CODE REVIEW

**Comment by mikem8361** [comment]
@kangaroo and @jkotas  would you mind review?

**Comment by jkotas** [comment]
LGTM

**Comment by kangaroo** [comment]
The logic in general looks ok to me.  I don't like popen("vmmap") from a security perspective though.  Its subject to a hijack if PATH is changed, and someone can drop an executable or script in $PATH.

Can we change that to /usr/bin/vmmap?

Also there are some tab/space nits.

File: src/pal/src/thread/process.cpp [hunk 1]
```diff
@@ -1886,57 +1872,104 @@ CreateProcessModules(
...
+
+        // Reading maps file line by line
+        while ((read = getline(&line, &lineLen, vmmapFile)) != -1)
         {
-            if (inode != 0)
+            void *startAddress, *endAddress;
+            char moduleName[PATH_MAX];
+            int size;
+
+            if (sscanf(line, "__TEXT %p-%p [ %dK] %*[-/rwxsp] SM=%*[A-Z] %s\n", &startAddress, &endAddress, &size, moduleName) == 4)
```
Comment (kangaroo) [inline src/pal/src/thread/process.cpp]: Its also worth noting that a hijacked vmmap could corrupt the stack here I think.

Comment (mikem8361) [inline src/pal/src/thread/process.cpp]: Is there a safe version of sscanf or better function to use on OSx (and Linux)?

mikem

**Comment by mikem8361** [comment]
I’ll fix that.

File: src/pal/src/thread/process.cpp (hunk 1 above)
Comment (kangaroo) [inline src/pal/src/thread/process.cpp]: You could add a format specifier for the length of moduleName to bound it to MAX_PATH-1 (for null) ?

**Pull request merged**